src/main/java/io/supertokens/Main.java

```diff
@@ -345,10 +346,11 @@ public void proceedToEnableFeatureFlag() {
 
     private void createDotStartedFileForThisProcess() throws IOException {
         CoreConfig config = Config.getBaseConfig(this);
+        String fileLocation = CLIOptions.get(this).getTempDirLocation() == null ? CLIOptions.get(this).getInstallationPath() : CLIOptions.get(this).getTempDirLocation();
```

Comment:
Not related to this file. CoreConfig is also using getInstallationPath() to figure out path for config.yaml and log file paths. Please see all places where we might finding a directory to write to.

Reply:
This is a special case for docker readonly fs. We are not intending to overwrite the installation path, or any of it. In the places where those are used, if those values (so, config location, log location) are not set, then it's a misconfigured core. The docker entrypoint sets those values as required. 
Moreover, If we automatically set the config location to the temp-dir-location, that could cause issues for example when the user intends to provide his/her own config.yml and also set a temp-dir-location (in docker)